src/blobs.cpp

```diff
@@ -66,10 +66,36 @@ namespace rhost {
             }
         }
 
-        bool to_blob(SEXP sexp, rhost::util::blob& blob) {
+        bool to_blob(SEXP sexp, std::vector<char>& blob) {
             bool result = false;
             rhost::util::errors_to_exceptions([&] {result = to_blob_internal(sexp, blob); });
             return result;
         }
+
+        void append_from_file(blob& blob, const char* path) {
```

Comment:
This is not new code, it just got moved from one file to another. 

As far as encoding handling, it's whatever the system's non-multibyte FS encoding is. This would be `CP_ACP` on Windows, and whatever the current locale specifies as encoding (normally UTF-8) on Unix. It's also consistent with how we handle file paths elsewhere in host code (you can see the call site for this uses `make_preferred`, which will use the right encoding).

We might want to review this later, but it'd have to be a sweep over the entire code, not just here.